scripts/bash_utils.sh

```diff
@@ -27,3 +27,22 @@ function setup_mason {
   fi
 
 }
+
+function setup_pre_commit {
+  echo "Installing pre-commit"
+  if [[ $OS = "Linux" ]] ; then
+    pip install pre-commit
+  elif [[ ${OS} = "Darwin" ]] ; then
+      if [[ $(command -v brew) == "" ]]; then
+        # Install Homebrew
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+      else
+        # Install pre-commit
+        brew install pre-commit
+      fi
+  else
+    pip install pre-commit
+  fi
+  echo "Setting up pre-commit hooks"
+  pre-commit install
+}
```

Comment:
to make it easier on ourselves I'd suggest
```suggestion
function setup_pre_commit {
  echo "INFO: Installing pre-commit"
  if [[ $(command -v python3) != "" ]]; then 
    python3 -m pip install pre-commit
  elif [[ $(command -v python) != "" ]]; then
    python -m pip install pre-commit
  else
    echo "WARNING: install python3 to set up pre-commit hooks."
  fi
  echo "INFO: Setting up pre-commit hooks"
  pre-commit install
}
```

or is there a particular reason why pip shouldn't be used on macs?

Reply:
Could be possible (for folks like me who don't use Python) that people have `brew` installed but not python ?

But it's fine to use `python`, I can install and test it myself, I (personally) prefer to install everything via `brew` instead of `python` or `cargo` so that there's only 1 package manager on the system for system-wide used libraries.